Meta/lint-ports.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+import os
+import re
+
+
+PORT_TABLE_REGEX = re.compile('^\\| \\[`([^`]+)`\\][^`]+$')
+PORT_TABLE_FILE = 'AvailablePorts.md'
+IGNORE_FILES = {'.gitignore', '.port_include.sh', PORT_TABLE_FILE, 'build_all.sh', 'ReadMe.md'}
+
+
+def read_port_table(filename):
+    ports = set()
+    with open(filename, 'r') as fp:
+        for line in fp:
+            match = PORT_TABLE_REGEX.match(line)
+            if not match:
+                continue
+            ports.add(match.groups()[0])
```

Comment:
nit: `match.group(1)`

But since it's a single group, I think the whole loop body can likely be replaced with `ports.add(PORT_TABLE_REGEX.findall(line))`

Reply:
If you use `re.M` in options, I think you don't even need to go line-by-line. Just

```
  return set(PORT_TABLE_REGEX.findall(open(filename, 'r').read()))
```

might be enough as function body.

Reply:
I'll use `with open()`, to avoid leaking the fd. It doesn't really matter here, but [I like cleaning up fds](#5342). :)